File: C++/0692. Top K Frequent Words.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        map<string, int> dict;
        for(auto w:words) {
            dict[w]++;            
        }
        vector<string> arr;

        for(auto x:dict) {
            arr.push_back(x.first);
        }

        sort(arr.begin(), arr.end(), [&dict](string x, string y) {if (dict[x] != dict[y]) return dict[x] > dict[y]; return x < y;});

        arr.erase(arr.begin() + k, arr.end());

        return arr;
    }
};

File: Python3/0692. Top K Frequent Words.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        dicts = collections.defaultdict(int)

        for w in words:
            dicts[w] += 1
        
        arr = sorted(dicts.keys(), key = lambda x:(-dicts[x], x))

        return arr[:k]
